Source/console-checkers/Source/ConsoleGameDisplay.cpp

```diff
@@ -0,0 +1,191 @@
+//---------------------------------------------------------------
+//
+// ConsoleGameDisplay.cpp
+//
+
+#include "ConsoleGameDisplay.h"
+
+#include "IGameBoardViewStrategy.h"
+#include "IGameStateDisplayInfo.h"
+#include "UIEvents.h"
+#include "UITextStrings.h"
+
+#include <iostream>
+#include <ranges>
+
+#include <spdlog/spdlog.h>
+// Magic include needed for spdlog to log a custom type.
+// Must be included after spdlog.h. Thanks spdlog.
+#include <spdlog/fmt/ostr.h>
+// Magic include needed for spdlog to log a vector inline.
+#include <spdlog/fmt/ranges.h>
+
+namespace Checkers {
+
+//===============================================================
+
+void ConsoleGameDisplay::Initialize(IGameStateDisplayInfo* displayData, UIEvents& events)
+{
+	m_gameStateInfo = displayData;
+
+	events.GetWelcomePromptRequestedEvent().subscribe(
+		[this]()
+	{
+		std::string uiScreenText = fmt::format("{}\n", UIText::s_welcomeMessage);
+		std::cout << uiScreenText;
+	});
+
+	events.GetGameBoardViewStrategyChangedEvent().subscribe(
+		[this](IGameBoardViewStrategy* view)
+	{
+		if (!view)
+		{
+			// We'll silently (to the player) fail here.
+			spdlog::error("UI notified of null view style change.");
+			return;
+		}
+
+		IGameBoardViewStrategy* old = m_gameBoardView;
+		m_gameBoardView = view;
+
+		// We assume that if we didn't have a view before, this call was initialization
+		// and not a change.
+		if (old)
+		{
+			// We only want to reprint the board if our view actually changed
+			std::cout << GetFullGameDisplayText();
+		}
+	});
+
+	events.GetPiecePromotedEvent().subscribe(
+	[this]()
+	{
+		std::cout << fmt::format("{} Kings a piece!\n", m_gameStateInfo->GetTurnPlayerId());
+	});
+
+	events.GetHelpPromptRequestedEvent().subscribe(
+		[this]()
+	{
+		std::cout << fmt::format(UIText::s_helpCommandPromptMessage, m_gameBoardView->GetMoveCommandSyntax());
+	});
+
+	events.GetCommandErrorPromptRequestedEvent().subscribe(
+		[this](const std::string& promptMessage)
+	{
+		std::cout << promptMessage << "\n";
+	});
+
+	events.GetTurnChangedEvent().subscribe(
+		[this]()
+	{
+		const std::string gameUI = fmt::format("\n{}\n{}",
+		"-------------------------------------",
+		GetFullGameDisplayText());
```

Comment:
Boom. It took me 5mins+ to find your "render" call. Your world/mental model for a game should be

Pseudo code:
```
while(!shutdown)
{
    HandleInputs();
    UpdateGameState();
    Render();
}
```

Then work back from there to flesh out the details on how all the systems will interact _to achieve the pseudo code above_

That's why I'm saying this feels like Checkers the UI. Even if this was a UI program this would be an odd place to have "Paint()"